plugins/node/opentelemetry-instrumentation-runtime/src/instrumentation.ts

```diff
@@ -0,0 +1,206 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { InstrumentationBase } from '@opentelemetry/instrumentation';
+import { RuntimeInstrumentationConfig } from './types';
+import { VERSION } from './version';
+import {
+  EventLoopUtilization,
+  IntervalHistogram,
+  monitorEventLoopDelay,
+  performance,
+} from 'perf_hooks';
+import {
+  hrTime,
+  hrTimeDuration,
+  hrTimeToMilliseconds,
+} from '@opentelemetry/core';
+import { AttributeNames } from './enums/AttributeNames';
+
+/**
+ * Runtime instrumentation for Opentelemetry
+ */
+export class RuntimeInstrumentation extends InstrumentationBase {
+  private eventLoopDelayHistogram: IntervalHistogram;
+  private ELU: EventLoopUtilization;
+  private _lastIntervalELU: EventLoopUtilization;
+  private config: RuntimeInstrumentationConfig;
+  private interval: NodeJS.Timeout | undefined;
+
+  constructor(
+    config: RuntimeInstrumentationConfig = {
+      monitorEventLoopDelayResolution: 10,
+      monitorEventLoopUtilizationResolution: 30000,
+    }
+  ) {
+    super('@opentelemetry/instrumentation-runtime', VERSION, config);
+
+    // https://nodejs.org/api/perf_hooks.html#perf_hooksmonitoreventloopdelayoptions
+    this.eventLoopDelayHistogram = monitorEventLoopDelay({
+      resolution: config.monitorEventLoopDelayResolution,
+    });
+    this.eventLoopDelayHistogram.enable();
+
+    // https://nodejs.org/api/perf_hooks.html#performanceeventlooputilizationutilization1-utilization2
+    const initialELU = performance.eventLoopUtilization();
+    this._lastIntervalELU = initialELU;
+    this.ELU = initialELU;
+    this.config = config;
+    this.enable();
```

Comment:
Review please! I need a place where I can set up an interval that
- Will only be created once
- Clear it (when needed)

The best matches I could find was "enable" and "disable" methods. LMK if there is something better available.

Reply:
I think, it would be better to use a [multiple-instrument callback](https://opentelemetry.io/docs/specs/otel/metrics/api/#multiple-instrument-callbacks). Removes the need to use an interval:

https://github.com/open-telemetry/opentelemetry-js/blob/1a8652aa5466510d2df2a232a0c8aa78857619c4/packages/sdk-metrics/README.md?plain=1#L52-L59

```ts
const loopUtilizationGauge = this.meter
  .createObservableGauge(AttributeNames.NODE_EVENT_LOOP_UTILIZATION, {
    description: 'The percentage utilization of the event loop.',
    unit: 'percent',
  });

const loopIdleGauge = this.meter
  .createObservableGauge(AttributeNames.NODE_EVENT_LOOP_UTILIZATION_IDLE, {
    description: 'The idle time utilization of event loop.',
    unit: 'ms',
  });

const loopActiveGauge = this.meter
  .createObservableGauge(
    AttributeNames.NODE_EVENT_LOOP_UTILIZATION_ACTIVE,
    {
      description: 'The active time utilization of event loop.',
      unit: 'ms',
    }
  );
  
this.meter.addBatchObservableCallback(async (observable) => {
  // Store the current ELU so it can be assigned later.
  const currentIntervalELU = performance.eventLoopUtilization();
  // Calculate the diff between the current and last before sending.
  const ELU = performance.eventLoopUtilization(
    currentIntervalELU,
    this._lastIntervalELU
  );
  observable.observe(ELU.utilization * 100);
  observable.observe(ELU.idle);
  observable.observe(ELU.active);
  // Assign over the last value to report the next interval.
  this._lastIntervalELU = currentIntervalELU;
}, [loopUtilizationGauge, loopIdleGauge, loopActiveGauge]);
```

Reply:
Yes, that is much nicer, then I can skip the config setting of modifying the window (30000ms) and just leave it to how often observable callback is invoked. Less complexity for the user.